website_product_subscription/models/subscription_request.py

```diff
@@ -60,12 +61,22 @@ def create_subscriber_web_access(self):
 
     @api.model
     def cron_create_scheduled_gift_user(self):
-        today = fields.Date.today()
+        now = fields.Datetime.from_string(fields.Datetime.now()).time()
+        if now < datetime.time(7, 0, 0) or now > datetime.time(22, 0, 0):
+            # Don't send e-mails during hours meant for sleep.
+            _logger.info(
+                "Skipping cron_create_scheduled_gift_user because it's late: %s"
+                % now
+            )
+            return
```

Comment:
Was that a request from wilfried ?

With this code + the condition `("gift_date", "=", fields.Date.today()),`, gift requests

-  submitted today after 22:00
- with gift_date = today

Will never be sent.

Reply:
Your best bet would probably be `gift_date between now - timedelta(hours=24) and now`

Reply:
:thinking: you're right. It wasn't a request from Wilfried, but in decreasing the interval in the cron, I figured that it'd be courteous to do this.

Following options:

1. Decrease the interval back to once per day and hope that that's sufficient.
2. Remove the above check and accept that e-mails are going to be sent at 01:00 at night.
3. Keep everything as-is, but change the condition back to `("gift_date", "<=", today)`, and sanitise the DB (i.e., disable/cancel/delete gift requests that are too old) before pushing this to prod.

Reply:
I think you should remove the night tests indeed: KISS. Did you see my time window suggestion?

Reply:
Did not see time window suggestion. Was it in the cron video?

Reply:
In my comment just above : `gift_date between now - timedelta(hours=24) and now`

Reply:
Not sure how to do that (is the above code or pseudocode?), but it's probably no longer necessary. I've reverted this commit to sending e-mails once per day, including gifts from the past.

Cath will be able clear the backlog before we push to prod, so the workaround is not needed.